implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalFxNamingConvention.java

```diff
@@ -81,21 +73,14 @@ public String name(String name, Meter.Type type, @Nullable String baseUnit) {
     public String tagKey(String key) {
         String conventionKey = delegate.tagKey(key);
 
-        conventionKey = START_UNDERSCORE_PATTERN.matcher(conventionKey).replaceAll(""); // 2
-        conventionKey = SF_PATTERN.matcher(conventionKey).replaceAll(""); // 2
-
-        conventionKey = PATTERN_TAG_KEY_DENYLISTED_CHARS.matcher(conventionKey).replaceAll("_");
-        if (!START_LETTERS_PATTERN.matcher(conventionKey).matches()) { // 3
+        if (conventionKey.startsWith("sf_") || !isAlphabet(conventionKey.charAt(0))) {
```

Comment:
This does not generate the same output as before. For example in case of `START_UNDERSCORE_PATTERN` that was just removed not adding an "a", same for `SF_PATTERN`. Do I miss anything?

Reply:
Yes, that is right. And I tried to lay that down explicitly in the PR description that this PR alters some of the existing behavior for better memory optimization. If you guys have a better approach to this, I would be more happy to wait for that implementation.

More about this is described in this issue - https://github.com/micrometer-metrics/micrometer/issues/3745

Reply:
I don't have a "better" approach, but I do have a recommendation to try to keep the current behavior if possible. For example by replacing the use of `SF_PATTERN.matcher(conventionKey).replaceAll("")` with something equivalent with less memory usage like:

```java
private static final SF_PREFIX = "sf_";
....

if (conventionKey.startsWith(SF_PREFIX)) {
  conventionKey = conventionKey.substring(SF_PREFIX.length())
}
```

Reply:
@bogdandrutu I fixed the code to retain existing behavior. I used recursive matching to eliminate any "sf_" and "_" as prefixes.
The basic idea behind this change is that the tag key convention should be as simple as possible when the key adheres to the SignalFx convention. Please review the changes and feel free to comment on this.